src/models/tools/geometry/jxg-board.ts

```diff
@@ -1,13 +1,15 @@
 import { JXGChange, JXGChangeAgent } from "./jxg-changes";
 import "./jxg";
 import { assign, each } from "lodash";
+import * as uuid from "uuid/v4";
 
 export const isBoard = (v: any) => v instanceof JXG.Board;
 
 export const boardChangeAgent: JXGChangeAgent = {
   create: (boardDomId: JXG.Board|string, change: JXGChange) => {
     const domElementID = boardDomId as string;
     const defaults = {
+            id: uuid(),
```

Comment:
What exactly is the deal with UUIDs here and all over the place?

Reply:
We need to make sure that every element has a UUID. Ideally, it's specified in the model/content but in case it's not, we add it here. So the model/content API adds the UUID if that API is used, but the jxg- API will add it if the model/content layer didn't. One place this could arise (temporarily) is with documents created before the introduction of the UUID, in which case the model/content won't have the UUID.